src/ssa/local_ssa.h

```diff
@@ -32,11 +34,14 @@ class local_SSAt
   inline local_SSAt(
     const goto_functiont &_goto_function,
     const namespacet &_ns,
+    const ssa_heap_analysist &_heap_analysis,
```

Comment:
This analysis must run on goto_functionst (since it is interprocedural) and only goto_functiont is passed to local_SSAt. Thus we need to run it before and pass the results of analysis to each function.